### DISCUSSION TITLE
PCL 源代码许可证和版本详解（2024年12月25日）

### DISCUSSION THREAD

**Discussion by Silverteal** [Meloong-Git/PCL#3933] [讨论]
以下内容是我，根据现有的启动器内/爱发电/GitHub仓库的各类版权信息整理的，没有经过作者的校对，不作为法律依据，不构成法律建议。

随着完整可编译代码正式发布，PCL2 可用的主程序版本现在可以分为三种：正式版，快照版，其他自行编译的版本。

# 权利相关：

**即使是最新的 PCL2 源代码许可证依然不满足 OSI 关于开源软件的定义，所以从称其为“开源”软件仍然是不适当的。**

所有版本的代码都可以用于“少量借鉴”
（什么是少量借鉴？参见《中华人民共和国著作权法》第二十四条，我敢说比任何其他解释都权威。）

而发布魔改版，简单来说，需要满足以下条件：
**_链接署名，责任自负，许可一致，保留原名，不做主题，不整回声，善意使用_**

# 用户相关
除快照版外，所有版本都可以被任何人使用和用于制作整合包、衍生作品
快照版可用于使用和制作其他形式的衍生作品（如视频），不能用于制作整合包

# 功能相关

所有版本都可以以完全的配置选项启动游戏

正式版，快照版可以在线更新，并接收网络公告

正式版，快照版可以获取 PCL2 特供下载版本列表

正式版，快照版可以使用回声洞，自定义下载，今日人品，千万别点等百宝箱功能

正式版，快照版可以下载 CurseForge 模组

快照版可以使用主题功能

# 源码相关

除了以下文件进行了修改外，本仓库的源代码应该与 PCL2 的发布版本相同。
  ```
  ModSecret.vb
  ModSetup.vb
  PageOtherTest.xaml
  PageOtherTest.xaml.vb
  ```

# Trivia
PCL 目前的许可证有一个小问题：
当许可证更新时，更新前的版本是遵循旧的许可证还是新的许可证？这不是“按理来说”就能确定的（就这一点来说，分歧会很大），需要明确的界定。不过考虑到“最终解释权”的存在，倒也不太会造成什么大不了的问题就是了。

**Comment by ZerkyLiu** [comment]
### 《中华人民共和国著作权法》第二十四条：

第二十四条　在下列情况下使用作品，可以不经著作权人许可，不向其支付报酬，但应当指明作者姓名或者名称、作品名称，并且不得影响该作品的正常使用，也不得不合理地损害著作权人的合法权益：

（一）为个人学习、研究或者欣赏，使用他人已经发表的作品；

（二）为介绍、评论某一作品或者说明某一问题，在作品中适当引用他人已经发表的作品；

（三）为报道新闻，在报纸、期刊、广播电台、电视台等媒体中不可避免地再现或者引用已经发表的作品；

（四）报纸、期刊、广播电台、电视台等媒体刊登或者播放其他报纸、期刊、广播电台、电视台等媒体已经发表的关于政治、经济、宗教问题的时事性文章，但著作权人声明不许刊登、播放的除外；

（五）报纸、期刊、广播电台、电视台等媒体刊登或者播放在公众集会上发表的讲话，但作者声明不许刊登、播放的除外；

（六）为学校课堂教学或者科学研究，翻译、改编、汇编、播放或者少量复制已经发表的作品，供教学或者科研人员使用，但不得出版发行；

（七）国家机关为执行公务在合理范围内使用已经发表的作品；

（八）图书馆、档案馆、纪念馆、博物馆、美术馆、文化馆等为陈列或者保存版本的需要，复制本馆收藏的作品；

（九）免费表演已经发表的作品，该表演未向公众收取费用，也未向表演者支付报酬，且不以营利为目的；

（十）对设置或者陈列在公共场所的艺术作品进行临摹、绘画、摄影、录像；

（十一）将中国公民、法人或者非法人组织已经发表的以国家通用语言文字创作的作品翻译成少数民族语言文字作品在国内出版发行；

（十二）以阅读障碍者能够感知的无障碍方式向其提供已经发表的作品；

（十三）法律、行政法规规定的其他情形。

https://www.gov.cn/guoqing/2021-10/29/content_5647633.htm

**Comment by Silverteal** [comment]
已经修改此Discussion的正文以符合最新版本的LICENSE
2024年7月8日

**Comment by Silverteal** [comment]
已经修改此Discussion的正文以符合最新版本的LICENSE
2024年12月25日

**Comment by LinQingYuu** [comment]
~现在有社区版了（bushi~
(无恶意)

**Reply by 3gf8jv4dv** [reply]
社区版不是官方版本，属于「自行修改后编译的版本」。

**Reply by Silverteal** [reply]
感谢您的提醒，由于现在仓库版和魔改版已经没有许可证上的差异，已删去这部分。

**Comment by CreeperIsASpy** [comment]
@LTCatt 考虑 Pin 一下？我觉得做的还不错.png
